examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
+Author: [Hazem Essam](https://twitter.com/hazemessamm) and [Santiago L. Valdarrama](https://twitter.com/svpino)
+Date created: 2021/03/13
+Last modified: 2021/03/22
+Description: Siamese network with tf.data and custom training loop.
```

Comment:
Suggestion: add a description of the task to help the user understand why they'd want to check this example

```suggestion
Description: A Siamese Network for a similarity comparison with `tf.data` and a custom training loop.
```

(Similar to https://keras.io/examples/vision/image_classification_from_scratch/ and similar)